CSharpAdvanceMasters/Assignment_6/Program.cs

```diff
@@ -0,0 +1,178 @@
+﻿using Spectre.Console;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Assignment_06
+{
+    public class Program
+    {
+        static async Task Main(string[] args)
+        {
+            bool endApp = false;
+
+            while (!endApp)
+            {
+                AnsiConsole.Write(new FigletText($"Threading").Centered().Color(Color.Aqua));
+
+                var crcObjectLisSelection = AnsiConsole.Prompt(
+                    new SelectionPrompt<string>()
+                        .PageSize(6)
+                        .Title("[yellow]Select list of object to view[/]")
+                        .MoreChoicesText("[grey](Move up and down to choose what to do)[/]")
+                        .AddChoices(new[]
+                        {
+                            $"1 - Sugar",
+                             "2 - Dakota",
+                             "3 - Chief",
+                             "4 - Ruby",
+                             "5 - Pegasus"
+                        }));
+
+                var crcUserOptionBet= crcObjectLisSelection.GetOption();
+                Console.WriteLine($"Selected list: {crcUserOptionBet.Item2}");
+                Console.WriteLine("\n");
+
+                // horse and race info
+                var randomSpeeds = await HorseRandomRaceSpeed();
+                var horseNames = HorseNames.ToList();
+                var raceInfo = GetHorseRaceInfo(randomSpeeds.ToList(), horseNames);
+
+                var winnerHorse = string.Empty;
+                var slowestHorse = string.Empty;
+
+                // thread
+                foreach (KeyValuePair<int, string> kvp in raceInfo)
+                {
+                    Thread horse = new(() => CountDown(kvp.Key, kvp.Value));
+                    horse.Name = kvp.Value;
+                    horse.Priority = Enum.Parse<ThreadPriority>(kvp.Key.ToString());
+                    horse.IsBackground = true;
+
+                    horse.Start();
+
+                    if (kvp.Key == (int)ThreadPriority.Highest) 
+                    {
+                        winnerHorse = kvp.Value;
```

Comment:
You can also use a shared state/shared variable/global variable to determine the winner. You can use a boolean if there's already a winner